### PR TITLE
Set labels directly on Unstructured

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

Previously, we were setting  labels directly on the `json.RawMessage`
object before unmarshaling it into an `unstructured.Unstructured`.
This commit replaces it to use the `SetLabels` method on the
Unstructured object. This means we do not have to escape special
characters in the label names anymore. In addition, we can remove the
`sjson` library since this was the only place where we were using it.

Signed-off-by: Dibyo Mukherjee <dibyo@google.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

**Comment by bobcatfish** [comment]
image

yaaaasss

File: pkg/apis/triggers/v1alpha1/register.go [hunk 1]
```diff
@@ -32,17 +32,8 @@ const (
...
-	TriggerBindingLabelKey = "/triggerbinding"
-
-	// TriggerTemplateKey is used as the label identifier for a TriggerTemplate
-	TriggerTemplateKey = "/triggertemplate"
-
 	// TriggerLabelKey is used as the label identifier for a Trigger
 	TriggerLabelKey = "/trigger"
-
-	// LabelEscape is an escaped GroupName safe for use in resource types.
-	LabelEscape = "tekton\\.dev"
```
Comment (bobcatfish) [inline pkg/apis/triggers/v1alpha1/register.go]: I'm a bit confused by these being deleted: 1) were we not using them before? 2) should we be?

File: Gopkg.lock [hunk 2]
```diff
@@ -484,38 +484,6 @@
...
-  version = "v1.0.0"
-
-[[projects]]
-  digest = "1:6b710a4259018ad39c57b847ef492762bcb705dc8d035647ba79abbf8bfacc29"
-  name = "github.com/tidwall/sjson"
-  packages = ["."]
-  pruneopts = "NUT"
-  revision = "25fb082a20e29e83fb7b7ef5f5919166aad1f084"
-  version = "v1.0.4"
-
```
Comment (bobcatfish) [inline Gopkg.lock]: :tada:

File: pkg/resources/create.go [hunk 3]
```diff
@@ -55,27 +54,24 @@ func FindAPIResource(apiVersion, kind string, c discoveryclient.ServerResourcesI
 		}
 		return r, nil
 	}
-	return nil, fmt.Errorf("Error could not find resource with apiVersion %s and kind %s", apiVersion, kind)
+	return nil, fmt.Errorf("error could not find resource with apiVersion %s and kind %s", apiVersion, kind)
```
Comment (bobcatfish) [inline pkg/resources/create.go]: :tada:

File: pkg/resources/create.go [hunk 4]
```diff
@@ -105,14 +101,13 @@ func Create(logger *zap.SugaredLogger, rt json.RawMessage, triggerName, eventID,
...
-		}
+func AddLabels(us *unstructured.Unstructured, labelsToAdd map[string]string) *unstructured.Unstructured {
+	labels := us.GetLabels()
+	for k, v := range labelsToAdd {
+		l := fmt.Sprintf("%s/%s", triggersv1.GroupName, strings.TrimLeft(k, "/"))
+		labels[l] = v
 	}
-	return rt, err
+
+	us.SetLabels(labels)
```
Comment (bobcatfish) [inline pkg/resources/create.go]: cool, nice find!!

**Comment by bobcatfish** [comment]
> everything looks great to me!

oh yeah except that deletion which I'm confused by XD

File: pkg/apis/triggers/v1alpha1/register.go (hunk 1 above)
Comment (dibyom) [reply pkg/apis/triggers/v1alpha1/register.go]: We needed it before when we were using sjson to set the labels...now we don't the escaping anymore 😄

**Pull request merged**